cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -689,9 +686,9 @@ runQueryStakeSnapshot (AnyConsensusModeParams cModeParams) network poolid = do
   eInMode <- toEraInMode era cMode
     & hoistMaybe (ShelleyQueryCmdEraConsensusModeMismatch (AnyConsensusMode cMode) anyE)
 
-  let qInMode = QueryInEra eInMode . QueryInShelleyBasedEra sbe $ QueryDebugLedgerState
+  let qInMode = QueryInEra eInMode . QueryInShelleyBasedEra sbe $ QueryStakeSnapshot $ Just $ Set.fromList mPoolIds
```

Comment:
So we have the following in consensus:
```
          case mPoolIds of
            Nothing ->
              StakeSnapshots
                { ssStakeSnapshots = getPoolStakes (Set.fromList (VMap.elems (SL._delegations _pstakeMark)))
                , ssMarkTotal      = getAllStake _pstakeMark
                , ssSetTotal       = getAllStake _pstakeSet
                , ssGoTotal        = getAllStake _pstakeGo
                }
            Just poolIds ->
              StakeSnapshots
                { ssStakeSnapshots = getPoolStakes poolIds
                , ssMarkTotal      = getAllStake _pstakeMark
                , ssSetTotal       = getAllStake _pstakeSet
                , ssGoTotal        = getAllStake _pstakeGo
                }
```
Should we also expose the ability to return the entire stake distribution by submitting `Nothing`?

Reply:
Done!  See PR description.